cluster/src/main/java/org/apache/iotdb/cluster/utils/ClusterUtils.java

```diff
@@ -368,6 +376,90 @@ public static PartitionGroup partitionByPathTimeWithSync(
+      logger.warn("the plan={} have no operator, ignore it", plan);
+      return;
+    }
+    switch (plan.getOperatorType()) {
+      case SET_STORAGE_GROUP:
+        majorVersion = currLogIndex;
+        minorVersion = 0;
+        plan.setMajorVersion(majorVersion);
+        plan.setMinorVersion(minorVersion);
+        break;
+      case DELETE_STORAGE_GROUP:
+        List<PartialPath> deletePathList = new ArrayList<>();
+        for (PartialPath storageGroupPath : plan.getPaths()) {
+          List<PartialPath> allRelatedStorageGroupPath =
+              IoTDB.metaManager.getStorageGroupPaths(storageGroupPath);
+          if (allRelatedStorageGroupPath.isEmpty()) {
+            throw new PathNotExistException(storageGroupPath.getFullPath(), true);
+          }
+          for (PartialPath path : allRelatedStorageGroupPath) {
+            StorageGroupMNode storageGroupMNode = IoTDB.metaManager.getStorageGroupNodeByPath(path);
+            majorVersion = storageGroupMNode.getMajorVersion();
+            minorVersion = storageGroupMNode.getMinorVersion();
+            path.setMajorVersion(majorVersion);
+            path.setMinorVersion(minorVersion);
+            deletePathList.add(path);
+          }
+        }
+        // replace the to be deleted paths
+        ((DeleteStorageGroupPlan) plan).setPaths(deletePathList);
+        break;
+      case CREATE_TIMESERIES:
+        PartialPath partialPath = ((CreateTimeSeriesPlan) plan).getPath();
+        StorageGroupMNode storageGroupMNode =
+            IoTDB.metaManager.getStorageGroupNodeByPath(partialPath);
+        majorVersion = storageGroupMNode.getMajorVersion();
+        minorVersion = currLogIndex;
+        ((CreateTimeSeriesPlan) plan).getPath().setMajorVersion(majorVersion);
+        ((CreateTimeSeriesPlan) plan).getPath().setMinorVersion(minorVersion);
+        break;
+      case ALTER_TIMESERIES:
+        partialPath = ((AlterTimeSeriesPlan) plan).getPath();
+        storageGroupMNode = IoTDB.metaManager.getStorageGroupNodeByPath(partialPath);
+        majorVersion = storageGroupMNode.getMajorVersion();
+        minorVersion = currLogIndex;
+        ((AlterTimeSeriesPlan) plan).getPath().setMajorVersion(majorVersion);
+        ((AlterTimeSeriesPlan) plan).getPath().setMinorVersion(minorVersion);
+        break;
+      case CREATE_ALIGNED_TIMESERIES:
+        partialPath = ((CreateAlignedTimeSeriesPlan) plan).getPrefixPath();
+        storageGroupMNode = IoTDB.metaManager.getStorageGroupNodeByPath(partialPath);
+        majorVersion = storageGroupMNode.getMajorVersion();
+        minorVersion = currLogIndex;
+        ((CreateAlignedTimeSeriesPlan) plan).getPrefixPath().setMajorVersion(majorVersion);
+        ((CreateAlignedTimeSeriesPlan) plan).getPrefixPath().setMinorVersion(minorVersion);
+        break;
+      case CREATE_MULTI_TIMESERIES:
+      case DELETE_TIMESERIES:
+        List<PartialPath> partialPaths = plan.getPaths();
+        for (PartialPath path : partialPaths) {
```

Comment:
why do not replace the path of plan?

Reply:
The path is get from the plan, so here the plan's path version have been chaned.